### PR TITLE
check backend configuration in compile

### PR DESCRIPTION
**Pull request by ajavadia**
### Summary
do a frontend check to make sure the qobj is compatible with backend.

### CODE REVIEW

**Comment by ajavadia** [comment]
I'll need to rebase this after #1386.

File: qiskit/tools/_compiler.py [hunk 1]
```diff
@@ -47,6 +48,15 @@ def compile(circuits, backend,
                       'Please pass an empty PassManager() instance instead',
                       DeprecationWarning)
 
+    if shots > backend.configuration().max_shots:
```
Comment (jaygambetta) [inline qiskit/tools/_compiler.py]: what happens when you using a state vector simulator.

Comment (ajavadia) [reply qiskit/tools/_compiler.py]: good point. I defined `max_shots` for the statevector simulators to be equal to 1. 

Therefore requesting more shots will raise an exception here.

**Comment by jaygambetta** [comment]
I dont know if i want to go down this path as then it ties compile more to a backend and then i cant run the qobj on another backend. Lets discuss after 0.7 so im going to put this on hold.

**Comment by delapuente** [comment]
This should be part of the backend validation, see **backend.validate** at #1047

**Comment by ajavadia** [comment]
Closing this as we want to keep the `compile()` function backend agnostic.

**Pull request closed without merging**